### PR TITLE
[QLN-46] feat: add report user of violation rules

### PR DESCRIPTION
**Pull request by lilinkw**
- Add report dialog
- Call API to BE
- Display result of report
image
image

### CODE REVIEW

File: messages/vi.json [hunk 1]
```diff
@@ -1657,5 +1657,61 @@
...
+        "cancel": "Hủy",
+        "submit": "Báo cáo"
+      },
+      "text_area": {
+        "placeholder": "Hãy nhập chi tiết vào đây"
+      }
+    },
+    "form_result_report": {
+      "report_success": "Báo cáo thành công",
+      "thank_you_reported": "Thank you for your contribute for better learning space",
```
Comment (blackmouse572) [inline messages/vi.json:1695]: Change this

File: src/app/[locale]/(main)/users/[id]/acitons/fetch-report-user.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  const option: RequestInit = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${token}`,
+    },
+    body: JSON.stringify({ accountId, reason }),
+    next: {
+      revalidate: 60, // Revalidate every 60 second
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/users/[id]/acitons/fetch-report-user.tsx:19]: POST dont cache

File: src/app/[locale]/(main)/users/[id]/components/report-button.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  return (
+    <Tooltip>
+      <TooltipTrigger asChild>
+        <Button onClick={hanldeClick} color="accent" isIconOnly>
+          <Icons.Report />
+        </Button>
+      </TooltipTrigger>
+      <TooltipContent>{content}</TooltipContent>
+    </Tooltip>
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/users/[id]/components/report-button.tsx:27]: use `NamedToolTip`

File: src/app/[locale]/(main)/users/[id]/components/report-user-dialog/report-user-result-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+          </div>
+          <div className="justify-center py-1 leading-5 text-black">
+            {i18n("form_result_report.received_submission")} &nbsp;
+            <span className="font-bold">{accountName}</span>&nbsp;
+            {i18n("form_result_report.of")} &nbsp;
+            <span className="font-bold">{i18n(reason?.text as any)}</span>&nbsp;
+            {i18n("form_result_report.against")} &nbsp;
+            <span className="underline">
+              {i18n("form_result_report.violation_rules")}
+            </span>
```
Comment (blackmouse572) [inline src/app/[locale]/(main)/users/[id]/components/report-user-dialog/report-user-result-dialog.tsx:63]: This is Link

**Review by blackmouse572 — CHANGES_REQUESTED** [review]

File: src/app/[locale]/(main)/users/[id]/components/report-user-dialog/report-user-result-dialog.tsx (hunk 4 above)
Comment (lilinkw) [reply src/app/[locale]/(main)/users/[id]/components/report-user-dialog/report-user-result-dialog.tsx:63]: Link to where?

**Review by blackmouse572 — APPROVED** [review]

**Pull request merged**